### PR TITLE
[FOSSA] Add agent plumbing code for Fossa, and allow users to connect to Fossa

### PR DESCRIPTION
**Pull request by angieyeh**
Task: NR-119842

We are adding the ability for users to connect to FOSSA in CodeStream

Screenshot 2023-06-27 at 3 25 09 PM
Screenshot 2023-06-27 at 3 21 33 PM

### CODE REVIEW

File: shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx [hunk 1]
```diff
@@ -36,11 +38,53 @@ export const CodeAnalyzers = (props: Props) => {
...
+	const [message, setMessage] = useState("");
+
+	const fetchCodeAnalysis = async () => {
+		if (loading) return;
+		setLoading(true);
+		if (!derivedState.currentRepo) {
+			setLoading(false);
+			return;
+		}
+		let msg;
```
Comment (angieyeh) [inline shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx:63]: this variable will be deleted and is only used for the purposes of making sure the client is receiving a message from the agent

File: shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx [hunk 2]
```diff
@@ -36,11 +38,53 @@ export const CodeAnalyzers = (props: Props) => {
...
+			console.error(error);
+		});
+	}, [
+		derivedState.currentRepo,
+		derivedState.providers,
+		derivedState.providerInfo,
+		props.paneState,
+	]);
+
+	const [loading, setLoading] = useState(false);
```
Comment (eric-j) [inline shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx:53]: minor thing, but we generally keep the useState declarations above hooks near the top of the component.

**Review by eric-j — APPROVED** [review]

**Review by dsellarsnr — APPROVED** [review]
Looks good!

File: shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx (hunk 2 above)
Comment (angieyeh) [reply shared/ui/Stream/CodeAnalyzers/CodeAnalyzers.tsx:53]: will update, thanks

**Pull request merged**